src/components/radio/radio_button.ts

```diff
@@ -0,0 +1,269 @@
+    });
+  }
+
+  @Input('value')
+  get value() {
+    return this.value_;
+  }
+
+  set value(value: any) {
+    if (this.value_ != value) {
+      ObservableWrapper.callEmit(this.change, null);
+    }
+    this.value_ = value;
+  }
+
+  @Input('selected')
+  get selected() {
+    return this.selected_;
+  }
+
+  set selected(selected: MdRadioButton) {
+    this.selected_ = selected;
+    this.value = selected.value;
+
+    selected.checked = true;
+  }
+
+  /** Registers a radio button with this group. */
+  register(radio: MdRadioButton) {
+    this.radios_.push(radio);
+  }
+}
+
+
+@Component({
+  selector: 'md-radio-button',
+  templateUrl: './components/radio/radio_button.html',
+  styleUrls: ['./components/radio/radio-button.css'],
+  encapsulation: ViewEncapsulation.None
+})
+export class MdRadioButton {
+  /** Input field wrapped reference. */
+  @ViewChild('input') private inputRef_: ElementRef;
+
+  /** Stores the raw input element. */
+  private inputElement_: HTMLInputElement;
+  @HostBinding('class.md-radio-focused') private isFocused_: boolean;
+
+  /** Whether this radio is checked. */
+  checked_: boolean;
+
+  /** The unique ID for the radio button. */
+  @HostBinding('id')
+  @Input()
+  id: string;
+
+  /** Analog to HTML 'name' attribute used to group radios for unique selection. */
+  @Input() name: string;
+
+  /** Whether this radio is disabled. */
+  disabled_: boolean;
+
+  /** Value assigned to this radio.*/
+  value_: any;
+
+  /** The parent radio group. May or may not be present. */
+  radioGroup: MdRadioGroup;
+
+  /** Dispatcher for coordinating radio unique-selection by name. */
+  radioDispatcher: MdRadioDispatcher;
+
+  /** Event emitted when the group value changes. */
+  @Output() change: EventEmitter<any>;
+
+  constructor(@Optional() @SkipSelf() @Host() radioGroup: MdRadioGroup, @Attribute('id') id: string,
+              radioDispatcher: MdRadioDispatcher) {
+    // Assertions. Ideally these should be stripped out by the compiler.
+    // TODO(jelbourn): Assert that there's no name binding AND a parent radio group.
+
+    this.radioGroup = radioGroup;
+    this.radioDispatcher = radioDispatcher;
+    this.value_ = null;
+    this.checked_ = false;
+    this.change = new EventEmitter();
+
+    this.id = isPresent(id) ? id : `md-radio-${_uniqueIdCounter++}`;
+
+    radioDispatcher.listen((name: string) => {
+      if (name == this.name) {
+        this.checked = false;
+      }
+    });
+
+    // When this radio-button is inside of a radio-group, the group determines the name.
+    if (isPresent(radioGroup)) {
+      this.name = radioGroup.name;
+      this.radioGroup.register(this);
+    }
+  }
+
+  ngAfterViewInit() {
+    this.inputElement_ = this.inputRef_.nativeElement;
+    this.inputElement_.addEventListener('change', this.onChange.bind(this));
+    this.inputElement_.addEventListener('focus', this.onFocus.bind(this));
+    this.inputElement_.addEventListener('blur', this.onBlur.bind(this));
```

Comment:
Instead of getting the raw input element and adding event listeners directly, it should just add events to the input in the template:
```html
<input type="radio" (change)="doSomething()" (focus)="doSomethingElse()">
```